frontend-react/src/components/shared/Table/Table.tsx

```diff
@@ -0,0 +1,225 @@
+import classnames from "classnames";
+import sanitizeHtml from "sanitize-html";
+import { useMemo, useState } from "react";
+import { Icon } from "@trussworks/react-uswds";
+
+import { convertStringToSortableRepresentation } from "../../../utils/misc";
+
+import styles from "./Table.module.scss";
+
+const FilterOptionsEnum = {
+    NONE: "none",
+    ASC: "asc",
+    DESC: "desc",
+} as const;
+
+interface SortableTableHeaderProps {
+    columnTitle: string;
+    activeColumn: string;
+    sortOrder: string;
+    setSortOrder: (order: string) => void;
+    setActiveColumn: (column: string) => void;
+}
+
+interface RowData {
+    columnKey: string;
+    content: string;
+    modalContent?: string;
+}
+
+export interface TableProps {
+    borderless?: boolean;
+    compact?: boolean;
+    emptyState?: boolean;
+    fullWidth?: boolean;
+    scrollable?: boolean;
+    sortable?: boolean;
+    stackedStyle?: "default" | "headers";
+    striped?: boolean;
+    rowData: RowData[][];
+}
+
+const SortableTableHeader = ({
+    columnTitle,
+    activeColumn,
+    sortOrder,
+    setSortOrder,
+    setActiveColumn,
+}: SortableTableHeaderProps) => {
+    let SortIcon = Icon.SortArrow;
+    const isActive = columnTitle === activeColumn;
+    if (isActive && sortOrder === FilterOptionsEnum.ASC) {
+        SortIcon = Icon.ArrowUpward;
+    } else if (isActive && sortOrder === FilterOptionsEnum.DESC) {
+        SortIcon = Icon.ArrowDownward;
+    }
+    const isCurrentlyActiveColumn = columnTitle === activeColumn;
+
+    const handleHeaderClick = () => {
+        if (!isCurrentlyActiveColumn) {
```

Comment:
I wonder if we can just maintain active column and sort order in a single state value:

```
{ <active_column_key>: <sort_order> }
```

It'll only be populated if there's an active column and current sort order; otherwise, `null` means that there's no sort order?